staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -226,9 +243,16 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 
 			start := strings.LastIndex(js[:syntax.Offset], "\n") + 1
 			line := strings.Count(js[:start], "\n")
-			return fmt.Errorf("json: line %d: %s", line, syntax.Error())
+			return JSONSyntaxError{
+				Line: line,
+				Err:  fmt.Errorf(syntax.Error()),
+			}
 		}
 	}
+
+	if err != nil && err != io.EOF && hasPrefix([]byte(err.Error()), []byte("yaml: line")) {
```

Comment:
> only occur when a call to gopkg.in/yaml.v2#Unmarshal results in an error.

Where does that happen?  That seems to be the natural place to type it. If you actually can't, you need to leave breadcrumbs in the comment so people don't think you were nuts.